spec/GeoRadius.coffee

```diff
@@ -0,0 +1,88 @@
+noflo = require 'noflo'
+redis = require 'redis'
+
+unless noflo.isBrowser()
+  chai = require 'chai'
+  path = require 'path'
+  baseDir = path.resolve __dirname, '../'
+else
+  baseDir = 'noflo-redis'
+
+describe 'GeoRadius component', ->
+  c = null
+  key = null
+  radius = null
+  unit = null
+  lat = null
+  lon = null
+  out = null
+  err = null
+  client = null
+  before (done) ->
+    @timeout 4000
+    loader = new noflo.ComponentLoader baseDir
+    loader.load 'redis/GeoRadius', (err, instance) ->
+      return done err if err
+      c = instance
+      key = noflo.internalSocket.createSocket()
+      c.inPorts.key.attach key
+      radius = noflo.internalSocket.createSocket()
+      c.inPorts.radius.attach radius
+      unit = noflo.internalSocket.createSocket()
+      c.inPorts.unit.attach unit
+      lat = noflo.internalSocket.createSocket()
+      c.inPorts.latitude.attach lat
+      lon = noflo.internalSocket.createSocket()
+      c.inPorts.longitude.attach lon
+      client = redis.createClient()
+      clientSocket = noflo.internalSocket.createSocket()
+      c.inPorts.client.attach clientSocket
+      clientSocket.send client
+      done()
+  after (done) ->
+    client.quit done
+  beforeEach ->
+    out = noflo.internalSocket.createSocket()
+    c.outPorts.out.attach out
+    err = noflo.internalSocket.createSocket()
+    c.outPorts.error.attach err
+  afterEach ->
+    c.outPorts.out.detach out
+    out = null
+    c.outPorts.error.detach err
+    err = null
+
+  describe 'getting members inside a radius', ->
```

Comment:
fbp-spec still not usable for component tests? What is blocking?

Reply:
Mainly because these components deal with a lot of side effects

Reply:
Also, no coverage support for fbp-spec yet, and that was the thing I was about to add next

Reply:
Are there problems with testing side-effects with fbp-spec that needs to be addressed?

Reply:
I think having some sort of "before" and "after" graphs might be nice

Reply:
Before can be done as part of the fixture (trigger the before actions an IIP, it will only run at fixture setup time). But indeed "after" cleanup does not have a good place to go.

Reply:
@jonnor want a feature request for fbp-spec?